File: app/src/main/java/com/pnuema/android/bible/data/firefly/Book.java
package com.pnuema.android.bible.data.firefly;

import com.pnuema.android.bible.data.IBook;

import androidx.annotation.NonNull;

@SuppressWarnings ("unused")
public class Book implements IBook {
    private int book_id;
    private String title;
    private boolean newTestament;

    @Override
    public int getId() {
        return book_id;
    }

    @NonNull
    @Override
    public String getName() {
        return title;
    }

    @Override
    public String getAbbreviation() {
        return null; //TODO if API is modified to return abbreviation this is here to support that
    }

    public boolean isNewTestament() {
        return newTestament;
    }
}

File: app/src/main/java/com/pnuema/bible/android/data/firefly/Book.kt
package com.pnuema.bible.android.data.firefly

import com.pnuema.bible.android.data.IBook
import com.pnuema.bible.android.database.BookOffline

data class Book(
    val book_id: Int,
    val title: String,
    val isNewTestament: Boolean
) : IBook {
    override fun getId() = book_id

    override fun getName() = title

    override fun getAbbreviation() = "" //TODO if API is modified to return abbreviation this is here to support that

    override fun convertToOfflineModel(version: String) = BookOffline(book_id = book_id, title = title, version = version, newTestament = isNewTestament)
}
